### ISSUE TITLE
Disabling debugInfo results in inability to re-order nodes.

### ISSUE THREAD

**Issue by joshuarh** [angular-ui-tree/angular-ui-tree#521]
When I disable Angular's debug info using the steps in [this Egghead tutorial](https://egghead.io/lessons/angularjs-new-in-angular-1-3-performance-boost-with-debuginfoenabled) in order to gain a performance boost, I lose the ability to drag and drop items. No information is logged in the console.

The following configuration option is what breaks the re-ordering:

```
module.config([ '$compileProvider', ($compileProvider) {
 $compileProvider.debugInfoEnabled(false);
}]);
```

Is `ui-tree` relying on attributes added by angular's debug info?

**Comment by StefanCodes** [comment]
@joshuabc that Egghead link is behind a paywall for me. Is the technique the same as the one documented here? https://docs.angularjs.org/guide/production

**Comment by joshuabc** [comment]
Hi @StefanCodes. Yes, I believe the technique is the same. I edited my comment with the exact code which results in the issue.

**Comment by StefanCodes** [comment]
Yup. I can reproduce this in my app, too.

Drag drop breaks using the snippet @joshuabc pasted above.
Running
````javascript
angular.reloadWithDebugInfo();
````
in the browser debugger makes drag drop start working again. I'll take a quick look, but this is outside my area of familiarity so I may not get anywhere.

**Comment by StefanCodes** [comment]
I think I see what's going on.

When you enable debugging, as per Angular docs:

> Where the compiler has created a new scope, the scope and either ng-scope or ng-isolated-scope CSS class are attached to the corresponding element. These scope references can then be accessed via element.scope() and element.isolateScope().

What this looks like in the rendered HTML is that the node objects (in my case, a div with the ui-tree-handle directive on it) are rendered with the ng-scope and ng-binding CSS classes, which I suspect is what Angular uses to get an object's scope using IAugmentedJQuery.scope() and .isolateScope(). As the Angular docs say, those scope functions are only available with debugging hints. And scope() is called twice in ui-tree - in the dragMove and the dragStart events!

I think the correct fix is to replace the calls to .scope() in both event handlers to a production-capable equivalent. Unfortunately I don't know Angular well enough to see a simple way to get a scope from an arbitrary DOM element (which is what the handlers are currently doing). Hopefully someone more experienced can suggest a solution.

The only workaround I'm aware of is "don't disable debug information on the compile provider". :(

**Comment by joshuabc** [comment]
Thanks for looking into it @StefanCodes!

**Comment by ernestopye** [comment]
+1 - just ran into this with my app. :(

**Comment by novgo** [comment]
+1 http://embed.plnkr.co/Tw51pOGImzlni6obV63X/preview

**Comment by Voles** [comment]
Related issue #425. I'll start working on this soon.

**Comment by Voles** [comment]
Please review: #596

**Comment by Voles** [comment]
Closing, let's have further discussion in the PR containing a fix (work in progress for now). See #648.

**Comment by Voles** [comment]
This issue has been fixed in v2.15.0.
